### PR TITLE
Add libpurple plugins for SIPE & WhatsApp & deploy them with other plugins

### PR DESCRIPTION
**Pull request by Herrie82**
Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

### CODE REVIEW

File: meta-luneos/recipes-support/pidgin/pidgin_2.10.12.bbappend [hunk 1]
```diff
@@ -1,3 +1,8 @@
 # We're only interested in the libpurple libraries and not the UI or
 # any other features so we keep those things disabled here.
 PACKAGECONFIG_remove = "consoleui"
+
+EXTRA_OECONF += " \
+--enable-plugins \
+--enable-debug \
```
Comment (shr-project) [inline meta-luneos/recipes-support/pidgin/pidgin_2.10.12.bbappend]: Do you want to keep debug enabled by default?

File: meta-luneos/recipes-support/pidgin/whatsapp-purple.bb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    oe_runmake CC="${CC}" CXX="${CXX}" EXTRA_INCLUDES="${TARGET_CFLAGS}" AR="${AR}";
+}
+
+do_install() {
+    oe_runmake DESTDIR="${D}" install;
+}
+
+FILES_${PN} += " \
+    ${libdir} \
+    ${datadir} \
```
Comment (shr-project) [inline meta-luneos/recipes-support/pidgin/whatsapp-purple.bb]: What paths are missing in default FILES_${PN}?

File: meta-luneos/recipes-support/pidgin/pidgin-sipe_1.21.1.bb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#PACKAGECONFIG[voice_and_video] = "--with-vv=yes,--with-vv=no,libnice gstreamer"
+PACKAGECONFIG[telepathy] = "--enable-telepathy=yes,--enable-telepathy=no,telepathy-glib gmime"
+#PACKAGECONFIG[gssapi_only] = "--enable-gssapi-only=yes,--enable-gssapi-only=no,krb5"
+PACKAGECONFIG[debug] = "--enable-debug=yes,--enable-debug=no,valgrind"
+
+FILES_${PN} += " \
+    ${libdir} \
+    ${datadir}/appdata \
+    ${datadir}/pixmaps \
+"
```
Comment (shr-project) [inline meta-luneos/recipes-support/pidgin/pidgin-sipe_1.21.1.bb]: What paths are missing in default FILES_${PN}? You shouldn't need to add libdir in most cases.

**Comment by shr-project** [comment]
3 comments inline

File: meta-luneos/recipes-support/pidgin/pidgin-sipe_1.21.1.bb (hunk 3 above)
Comment (Herrie82) [inline meta-luneos/recipes-support/pidgin/pidgin-sipe_1.21.1.bb]: I get the following when I don't provide a FILES_${PN}:

WARNING: pidgin-sipe-1.21.1-r0 do_package: QA Issue: pidgin-sipe: Files/directories were installed but not shipped in any package:
  /usr/share/appdata
  /usr/share/appdata/pidgin-sipe.metainfo.xml
  /usr/lib/purple-2/libsipe.so
  /usr/lib/purple-2/libsipe.la
Please set FILES such that these items are packaged. Alternatively if they are unneeded, avoid installing them or delete them within do_install.
pidgin-sipe: 4 installed and not shipped files. [installed-vs-shipped]

File: meta-luneos/recipes-support/pidgin/whatsapp-purple.bb (hunk 2 above)
Comment (Herrie82) [inline meta-luneos/recipes-support/pidgin/whatsapp-purple.bb]: I get the following when I omit libdir and/or datadir:

WARNING: whatsapp-purple-0.9.0+gitAUTOINC+07ed931a3d-r0 do_package: QA Issue: whatsapp-purple: Files/directories were installed but not shipped in any package:
  /usr/lib/purple-2/libwhatsapp.so
Please set FILES such that these items are packaged. Alternatively if they are unneeded, avoid installing them or delete them within do_install.
whatsapp-purple: 1 installed and not shipped files. [installed-vs-shipped]

File: meta-luneos/recipes-support/pidgin/pidgin_2.10.12.bbappend (hunk 1 above)
Comment (Herrie82) [inline meta-luneos/recipes-support/pidgin/pidgin_2.10.12.bbappend]: For the time being yes, we might eventually disable this.

**Comment by Herrie82** [comment]
Replied to comments

File: meta-luneos/recipes-support/pidgin/whatsapp-purple.bb (hunk 2 above)
Comment (shr-project) [inline meta-luneos/recipes-support/pidgin/whatsapp-purple.bb]: Is it symlink or actual elf library? symlink should be in PN-dev.

File: meta-luneos/recipes-support/pidgin/pidgin-sipe_1.21.1.bb (hunk 3 above)
Comment (shr-project) [inline meta-luneos/recipes-support/pidgin/pidgin-sipe_1.21.1.bb]: Is libsipe.so symlink or actual elf library? symlink should be in PN-dev. libsipe.la definitely belongs to PN-dev not PN

File: meta-luneos/recipes-support/pidgin/whatsapp-purple.bb (hunk 2 above)
Comment (Herrie82) [inline meta-luneos/recipes-support/pidgin/whatsapp-purple.bb]: Actual elf library.

File: meta-luneos/recipes-support/pidgin/pidgin-sipe_1.21.1.bb (hunk 3 above)
Comment (Herrie82) [inline meta-luneos/recipes-support/pidgin/pidgin-sipe_1.21.1.bb]: libsipe.so = elf library. 

I've updated based on your comments: https://github.com/webOS-ports/meta-webos-ports/pull/186/commits/6eb6543ee13abaab1be0d9d97a5a79d14dd32f31

Also added freeimage recipe from upstream repo required for whatsapp purple https://github.com/openembedded/openembedded/blob/fabd8e6d07d3cd0cc93c2a0fc804f8c8f316c649/recipes/ogre/freeimage_3.11.0.bb

File: meta-luneos/recipes-support/freeimage/freeimage_3.11.0.bb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+do_unpack[depends] += "unzip-native:do_populate_sysroot"
```
Comment (shr-project) [inline meta-luneos/recipes-support/freeimage/freeimage_3.11.0.bb]: Why is this needed? It should be automatic nowadays.

File: meta-luneos/recipes-support/freeimage/freeimage_3.11.0.bb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+PR = "r1"
```
Comment (shr-project) [inline meta-luneos/recipes-support/freeimage/freeimage_3.11.0.bb]: Drop

File: meta-luneos/recipes-support/freeimage/freeimage_3.11.0.bb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    oe_runmake INSTALLDIR="${D}${libdir}" INCDIR="${D}${includedir}" install
+}
+
+INSANE_SKIP_${PN} = "already-stripped"
+
+FILES_${PN} += "${libdir}"
+FILES_SOLIBSDEV = "${libdir}/lib${PN}.so"
+
+SRC_URI[md5sum] = "ad1db36414391417654ba7bf1c0277d3"
+SRC_URI[sha256sum] = "84021b8c0b86e5801479474ad9a99c18d121508ee16d363e02ddcbf24195340c"
```
Comment (shr-project) [inline meta-luneos/recipes-support/freeimage/freeimage_3.11.0.bb]: Please move nex to SRC_URI

File: meta-luneos/recipes-support/freeimage/freeimage_3.11.0.bb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+do_install() {
+    install -d ${D}${libdir}
+    install -d ${D}${includedir}
+    oe_runmake INSTALLDIR="${D}${libdir}" INCDIR="${D}${includedir}" install
+}
+
+INSANE_SKIP_${PN} = "already-stripped"
+
+FILES_${PN} += "${libdir}"
+FILES_SOLIBSDEV = "${libdir}/lib${PN}.so"
```
Comment (shr-project) [inline meta-luneos/recipes-support/freeimage/freeimage_3.11.0.bb]: This looks almost like the default value. why is it needed?

File: meta-luneos/recipes-support/freeimage/freeimage_3.17.0.bb [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    ${S}/Makefile.gnu
+}
+
+do_install() {
+    install -d ${D}${libdir}
+    install -d ${D}${includedir}
+    oe_runmake INSTALLDIR="${D}${libdir}" INCDIR="${D}${includedir}" install
+}
+
+INSANE_SKIP_${PN} = "already-stripped dev-so"
```
Comment (Herrie82) [inline meta-luneos/recipes-support/freeimage/freeimage_3.17.0.bb]: I updated to FreeImage 3.17.0 which needed a patch to get it working with GCC 5.1+. Not sure you're happy with the INSANE_SKIP_${PN} and the FILES here? I squashed all commits into this one now.
